internal/credential/vault/repository_credential_store.go

```diff
@@ -533,68 +521,73 @@ func (r *Repository) UpdateCredentialStore(ctx context.Context, cs *CredentialSt
 	}
 	cs = cs.clone()
 
-	var updateToken, updateClientCert, deleteClientCert bool
-	var clientCertPlaceholder []byte
+	updateToken := false
 	for _, f := range fieldMaskPaths {
 		switch {
-		case strings.EqualFold("Name", f):
-		case strings.EqualFold("Description", f):
-		case strings.EqualFold("Namespace", f):
-		case strings.EqualFold("TlsServerName", f):
-		case strings.EqualFold("TlsSkipVerify", f):
-		case strings.EqualFold("CaCert", f):
-		case strings.EqualFold("VaultAddress", f):
-		case strings.EqualFold("Token", f):
+		case strings.EqualFold(nameField, f):
+		case strings.EqualFold(descriptionField, f):
+		case strings.EqualFold(namespaceField, f):
+		case strings.EqualFold(tlsServerNameField, f):
+		case strings.EqualFold(tlsSkipVerifyField, f):
+		case strings.EqualFold(caCertField, f):
+		case strings.EqualFold(vaultAddressField, f):
+		case strings.EqualFold(certificateField, f):
+		case strings.EqualFold(certificateKeyField, f):
```

Comment:
Can you update the method doc regarding the ability to update the client certificate without updating the client certificate private key?